### PR TITLE
Add changelog.json for end user facing changes in KKP 2.22.0

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:

I've extracted what I considered the most important end user (i.e. someone who uses the dashboard of a KKP instance) facing changes and updated the `changelog.json` with it. I assume it does not cover everything, but unless I've missed something crucial it's probably a solid amount of information for users. The UI links to the full changelog anyway as well.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind documentation

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: modules/web/src/assets/config/changelog.json [hunk 1]
```diff
@@ -1,4 +1,185 @@
...
+    },
+    {
+        "category": "added",
+        "description": "Add Canal CNI v3.24."
+    },
+    {
+        "category": "added",
+        "description": "Add Cilium CNI v1.13.0."
+    },
+
```
Comment (pkprzekwas) [inline modules/web/src/assets/config/changelog.json:60]: Are you sure those empty lines don't affect JSON parsing?

Comment (embik) [reply modules/web/src/assets/config/changelog.json:60]: I tried it by running the dashboard locally and it was fine, I was able to see the full changelog via it.

**Comment by Waseem826** [comment]
I think we've missed following features:
- Default Project Quota
- Live estimation of Project Quota while creating cluster/MD.
- vSphere tag categories
- Cluster RBAC changes (maybe)
- Edit/Customize cluster templates
- Default application installation when Cilium is selected as CNI plugin

**Comment by embik** [comment]
> I think we've missed following features:
> 
> * Default Project Quota
> * Live estimation of Project Quota while creating cluster/MD.

Added that, but I suspect we should look into having changelog items only showing up for EE installations.

> * vSphere tag categories

I'm not sure what to write here to be honest since I don't fully understand the changes.

> * Cluster RBAC changes (maybe)

Not sure here either, sorry.

> * Edit/Customize cluster templates

Added.

> * Default application installation when Cilium is selected as CNI plugin

I think this is more of an internal change. It's still on the changelog, but I'm not sure it should be on the abridged version.

**Review by Waseem826 — APPROVED** [review]

**Pull request merged**